### ISSUE TITLE
Clean up token on deauthorize

### ISSUE THREAD

**Issue by AlexTugarev** [gitpod-io/gitpod#3800]
### Bug description

When disconnecting a provider, tokens should be deleted as well.

### Steps to reproduce

Go to `/integrations` page and disconnect a previously connected provider.

Related to https://github.com/gitpod-io/gitpod/issues/3754

Additional information
* on account deletion / reconnecting with the same account the tokens will be cleaned up ✅ 
* on connecting with another account (after previously disconnecting from first account) the old token remains ❌ 
  * in such cases, no tokens are returned (the DB interface and the `getToken` operation will throw an error,) thus unauthorized access is prevented ✅ 
  * further there is an re-authorization request in place when trying to open a workspace for the affected git provider, which will clean up the situation on success ✅  

**TL;DR** the UX is impacted as it looks like the authorization is missing ❌
